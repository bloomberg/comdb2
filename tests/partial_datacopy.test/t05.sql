create table t {
    schema {
        int a
        int b
        int c
        int d
    }
    keys {
        datacopy(a) "a" = a
    }
};$$

create table t2 {
    schema {
        int a
        int b
        int c
        int d
    }
    keys {
        datacopy(a, b, c, d) "abcd" = a + b + c + d
    }
};$$

select * from comdb2_keys where tablename='t';
select * from comdb2_partial_datacopies where tablename='t';

select * from comdb2_keys where tablename='t2';
select * from comdb2_partial_datacopies where tablename='t2';

insert into t with recursive r(n) as (select 1 union all select n + 4 from r where n < 97) select n, n + 1, n + 2, n + 3 from r;
select * from t where a < 40;
select a from t where a < 40;

insert into t2 with recursive r(n) as (select 1 union all select n + 4 from r where n < 97) select n, n + 1, n + 2, n + 3 from r;
select * from t2 where a < 40;

drop table t;
drop table t2;
