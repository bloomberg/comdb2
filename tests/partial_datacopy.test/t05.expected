(tablename='t', keyname='A', keynumber=0, isunique='Y', isdatacopy='N', isrecnum='N', condition=NULL, uniqnulls='N', ispartialdatacopy='Y')
(tablename='t', keyname='A', columnname='a')
(tablename='t2', keyname='ABCD', keynumber=0, isunique='Y', isdatacopy='N', isrecnum='N', condition=NULL, uniqnulls='N', ispartialdatacopy='Y')
(tablename='t2', keyname='ABCD', columnname='a')
(tablename='t2', keyname='ABCD', columnname='b')
(tablename='t2', keyname='ABCD', columnname='c')
(tablename='t2', keyname='ABCD', columnname='d')
(rows inserted=25)
(a=1, b=2, c=3, d=4)
(a=5, b=6, c=7, d=8)
(a=9, b=10, c=11, d=12)
(a=13, b=14, c=15, d=16)
(a=17, b=18, c=19, d=20)
(a=21, b=22, c=23, d=24)
(a=25, b=26, c=27, d=28)
(a=29, b=30, c=31, d=32)
(a=33, b=34, c=35, d=36)
(a=37, b=38, c=39, d=40)
(a=1)
(a=5)
(a=9)
(a=13)
(a=17)
(a=21)
(a=25)
(a=29)
(a=33)
(a=37)
(rows inserted=25)
(a=1, b=2, c=3, d=4)
(a=5, b=6, c=7, d=8)
(a=9, b=10, c=11, d=12)
(a=13, b=14, c=15, d=16)
(a=17, b=18, c=19, d=20)
(a=21, b=22, c=23, d=24)
(a=25, b=26, c=27, d=28)
(a=29, b=30, c=31, d=32)
(a=33, b=34, c=35, d=36)
(a=37, b=38, c=39, d=40)
